watertap/core/membrane_channel_base.py

```diff
@@ -83,16 +93,6 @@ class PressureChangeType(Enum):
     calculated = auto()
 
 
-class FrictionFactor(Enum):
```

Comment:
My only hesitation about complete removal of a friction_factor config option is that, although it appears cleaner now with your implementation, what happens in the future if someone wants to add a custom friction_factor relationship? 

On the one hand, we could consider adding a new ModuleType option--which would require provision of a new membrane area and friction factor calculation. 

On the other hand, if we wanted to just maintain a spiral wound type for the membrane area determination and only wanted to provide a custom friction factor relationship, we couldn't do that with the new changes. 

There might be other options that I didn't cover, but figured this might be the last issue to reflect on before removing separate friction_factor options completely.

Another thing we could consider is keeping friction_factor with modified options. FOr example, we could have a "default" or "default_by_module_type") option for friction_fractor configuration that would simply use the relationship associated with the module_type selected; i.e., the user would not need to specify "flat_sheet" or "spiral_wound" explicitly for setting up the friction_factor config. Additional config options for the friction_factor config would consider alternative relationships that might be added later on, or custom relationships provided on-the-fly.

Reply:
Another case where I think allowing for flexibility in adding in friction factor relationships is for counter or co-current modules like OARO for example. We may want to consider different friction factor equations for each side of the membrane (e.g., higher pressure drop on permeate side embodied by some equation that differs from the one used on the feed side). Keeping the friction factor configuration option would allow the user to make more detailed decisions. That said, a flaw in the existing implementation (before your changes) is that I think we only had a single friction factor config option intended for the feed side and didn't consider alternative relationships for the permeate side (when relevant, i.e., in the case of OARO).

Reply:
Well we could keep the friction_factor option purely as an option for the constraint used for pressure loss. The module_type option isn't meant to have anything to do with the pressure loss equations used, it's just to assure correct channel dimensions are used.

We can set the friction_factor option to a default, but then we're asking the user to implement their custom constraints on the unit model level and not their flowsheet level

Reply:
> The module_type option isn't meant to have anything to do with the pressure loss equations used, it's just to assure correct channel dimensions are used.

True, but currently, we happen to have two friction factor relationships that correspond to a specified flat sheet correlation and another corresponding to a classic spiral wound correlation. This is the only reason why I mentioned possibly having "default" depend on module_type.

Reply:
I support setting default_by_module as the default option for the friction_factor config

Reply:
> On the one hand, we could consider adding a new ModuleType option--which would require provision of a new membrane area and friction factor calculation.

I believe it's appropriate to define new area calculations for additional module types. Considering module types like tubular, capillary, and hollow fiber, all require area calculations based on the diameter of the tubes or fibers, it makes sense to introduce a new area calculation method tailored to these specifications.